src/frollR.c

```diff
@@ -235,6 +235,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
     Rprintf("%s: processing of %d column(s) and %d window(s) took %.3fs\n", __func__, nx, nk, omp_get_wtime()-tic);
 
   UNPROTECT(protecti);
+  free(dans);
```

Comment:
8 lines above we can terminate this function by calling `error`, I believe `free` has to be there too.

Reply:
Do we need to extract `dans[i*nk+j].message[3]` before `free(dans)`?

Like

```
char err_msg = dans[i*nk+j].message[3];
free(dans);
error("%s: %d: %s", __func__, i*nk+j+1, err_msg);
```

Reply:
I think so. Probably `char *err_msg` or `char err_msg[4096]` looking at types.h

Reply:
ok have a look

Reply:
if we put 4096 then best to have a macro defining that size, so the future change will not require updates in multiple places. If `err_msg*` works then it is probably better.

Reply:
Maybe we need `strcpy` here actually. @mattdowle ?

Reply:
actually it's not compiling... still investigating

Reply:
I would do a macro ANS_MSG_SIZE = 4096 in types.h, then used that here to define a new char array, and `strcpy` into it.

Reply:
see current commit...

I saw a reference for `strcpy`, but was getting a `warning: implicit declaration of function 'strcpy' not valid in C99` (or similar). Couldn't figure out why (I see other `str*` functions in the script and `#include <string.h>` didn't make a difference)...

Reply:
hmm, no ideas for now

Reply:
maybe it was indeed a typo. can take over from here? I'm off for the night

Reply:
maybe it was indeed a typo. can take over from here? I'm off for the night

Reply:
so maybe `memcpy` simply

Reply:
`strncpy` seems to work fine, tested on `verbose` output because `error` only happens for failed `malloc`.